sdk/java-sdk-spring/src/main/scala/kalix/javasdk/impl/eventsourcedentity/ReflectiveEventSourcedEntityRouter.scala

```diff
@@ -84,11 +91,18 @@ class ReflectiveEventSourcedEntityRouter[S, E, ES <: EventSourcedEntity[S, E]](
         commandContext.metadata())
 
     val inputTypeUrl = command.asInstanceOf[ScalaPbAny].typeUrl
-
-    commandHandler
+    val methodInvoker = commandHandler
       .getInvoker(inputTypeUrl)
-      .invoke(entity, invocationContext)
-      .asInstanceOf[EventSourcedEntity.Effect[_]]
+
+    val span = telemetry.buildSpan(entity, inputTypeUrl, Some(methodInvoker), commandContext.metadata)
+    try {
+      methodInvoker
+        .invoke(entity, invocationContext)
+        .asInstanceOf[EventSourcedEntity.Effect[_]]
+    } finally {
+      span.map(s => s.end())
```

Comment:
Is that really the span end, this is as soon as the user method returns, wouldn't it be after handling the returned effect, for example after persist and event has been applied? The effect could also be async and not have completed when the methor returns.

Reply:
went one level deeper, to the ActionImpl but still not enough. [Follow](https://github.com/lightbend/kalix-jvm-sdk/pull/1763#discussion_r1328376206).